### DISCUSSION TITLE
Cannot allow notification access - Pixel 6

### DISCUSSION THREAD

**Discussion by psonnera** [NightscoutFoundation/xDrip#2627] [Q&A]
Trying to get Companion app to follow CamAPS, Allow notification access is greyed out for xDrip+.

![image](https://user-images.githubusercontent.com/33809361/212171981-c1f71e83-e8fe-4396-973d-243e689dfa2b.png)
![image](https://user-images.githubusercontent.com/33809361/212172048-2de66a67-a960-4156-8509-a97876566136.png)
![image](https://user-images.githubusercontent.com/33809361/212172168-ee5aa030-6524-4e97-94e2-081cd592eacf.png)

**Comment by jamorham** [comment]
If you long press on the xDrip icon on the app drawer and select app info, is there a 3 dot menu in the top right of the app info page that allows you to enable restricted settings? If so enable that and try again. Otherwise is the phone running parental controls or other device management such as being a phone controlled by an employer?

**Comment by clixbya** [comment]
All it says is "All Permissions" and no I don't have any parental
restrictions on it as far as i am aware but i will check now.

**Comment by jamorham** [comment]
If you're not able to resolve it can you screenshot your android version page with which android version and build number being used here.

**Reply by clixbya** [reply]
![Screenshot_20230112-205752](https://user-images.githubusercontent.com/122568528/212180006-a1751881-d819-472c-82ec-441eb3178839.png)

**Comment by ostefek99** [comment]
I have had the same issue just today. You have to find the app (either on home screen or in the app menu) and long press -> app info -> look for three dots (vertical) in the right upper corner. Tap it and press the "Allow restricted settings". You should be then able to enable the access to device & app notifications.
I have spent couple of hours looking for the way how to enable it... It's not present in Android 12. It was introduced in Android 13 and specifically for sideloaded apps.

More info can be found [here](https://blog.esper.io/android-13-sideloading-restriction-harder-malware-abuse-accessibility-apis/).

**Comment by clixbya** [comment]
Something wrong with mine then cos mine doesnt say that all it says it "all
permissons" will delete everything and try it again without granting
permission to anything

**Comment by rudysroost** [comment]
You have to touch "All Permissions" do display the options.

**Comment by jamorham** [comment]
@clixbya are you able to resolve the problem with these suggestions?

**Comment by clixbya** [comment]
Nope

**Comment by jamorham** [comment]
This video shows a walk-through with another app - does this work? If not where do you see differences? https://www.youtube.com/watch?v=XSHTo8qwitc

**Reply by Rehellek** [reply]
Thanks a million, for this post jamorham. Saved me binning my son's new phone, he's delighted to get it running again. Xdrip a brilliant app. Thanks again.

**Comment by clixbya** [comment]
I dont get the 3 dots on the app info page and when i do it just says "all
permissions" when i click on it it just shows me everything that has
permission i cannot edit them

**Comment by jamorham** [comment]
I've tested this on `TQ1A.230105.001.A2` which should be pretty much identical and it works exactly as in the video. I'm wondering why your device is behaving differently.  If you go to android settings and search for `Device Admin Apps` does it show any active apps? 

If there is nothing active there, can you try going to `About phone` in settings and then tap on the `Build Number` section at the bottom 7 times until it activates developer mode and see if that changes anything?

**Comment by clixbya** [comment]
Still exactly the same, i did all the developer stuff aswell

**Comment by jamorham** [comment]
Please confirm your xDrip+ version in use and that it is the official Nightscout xDrip+ version from this repository? If so please also go in to xDrip and search the settings for `accessibility` and then click that preference item to try to enable the always on display feature. This is just a test to see if it will allow access to that.

**Comment by jamorham** [comment]
Additionally, if you tap on the slider for the notification access does it pop up anything saying the setting is restricted? If so that may be what then enables the 3 dot menu.

**Comment by clixbya** [comment]
![Screenshot_20230116-214333](https://user-images.githubusercontent.com/122568528/212770599-aebd3f00-6b30-445d-8406-e653d2f7470d.png)
![Screenshot_20230116-214340](https://user-images.githubusercontent.com/122568528/212770603-237eedf9-71c8-4ae3-a46e-8a812cf2618d.png)
![Screenshot_20230116-214347](https://user-images.githubusercontent.com/122568528/212770605-f887a129-7fa6-46bb-a4c9-b4c78b34c79e.png)
![Screenshot_20230116-214353](https://user-images.githubusercontent.com/122568528/212770609-4c9459ea-d895-493e-8f6d-1a27d2579d75.png)
![Screenshot_20230116-214445](https://user-images.githubusercontent.com/122568528/212770614-05dfe2d3-ccdf-43a8-84cc-773e38999bb2.png)
![Screenshot_20230116-214451](https://user-images.githubusercontent.com/122568528/212770616-9453567a-9e51-46de-a290-22e871feb8b0.png)
![Screenshot_20230116-214605](https://user-images.githubusercontent.com/122568528/212770619-3fe118c8-234a-478d-8d25-3abf712f9346.png)
![Screenshot_20230116-214703](https://user-images.githubusercontent.com/122568528/212770621-c0749b20-6d9f-49dd-8b34-0253a4e7c3de.png)
![Screenshot_20230116-214719](https://user-images.githubusercontent.com/122568528/212770624-ed8e6d7e-f4e5-4ad5-b0ea-ad6114c0c918.png)
This is what i get doing everything you have advised me to, ita just not having any of it

**Comment by drrobintravis** [comment]
Hi! I just bought a new Pixel 6s to use with camapsfx. I have previously used xdrip in companion mode on a Samsung. Now that I am using the Pixel I cannot get it to grant the necessary access to read system notifications. I am able to allow notification access, but at the bottom it says "this app does not allow enhanced settings". Are allowing 
![Screenshot_20240222-080118](https://github.com/NightscoutFoundation/xDrip/assets/113501662/355ce2dc-e73d-4d46-9ed3-cc04dc1e2992)
system notifications hidden for me thus making it impossible to use companion mode?

**Reply by Navid200** [reply]
When that happens, go to Android Settings &#8722;> Apps &#8722;> xDrip 

You should see a 3-dot menu at the top right.
There should be "Allow restricted settings".
